src/net_processing.cpp

```diff
@@ -1577,6 +1579,20 @@ void PeerManagerImpl::FinalizeNode(const CNode& node)
     LogPrint(BCLog::NET, "Cleared nodestate for peer=%d\n", nodeid);
 }
 
+bool PeerManagerImpl::HasDesirableServiceFlags(ServiceFlags services)
```

Comment:
> The comment ("shortcut for...") was helpful for me, any reason to drop it?

That was because the method was moved to the net interface ([see](https://github.com/bitcoin/bitcoin/blob/1428b7f4e8c648fdc68ba5744d640d120d5d78eb/src/net.h#L673)). And there, this is no longer a shortcut. It is the only available method.

Reply:
I interpreted the comment as explaining the not-so-trivial bit-operator logic:
`!(GetDesirableServiceFlags(services) & (~services));` 
is still a shortcut for the longer, but more understandable alternative
`services & GetDesirableServiceFlags(services)) == GetDesirableServiceFlags(services)`
no matter where the function is situated.

Reply:
Ok. For that, we can locate the comment on the implementation (above the no-so-trivial logic). So it's clear what that single line does. I just don't think that is good to place it at the API level. Where `GetDesirableServiceFlags` doesn't exist and `HasDesirableServiceFlags()` could be implemented in different manners.

Reply:
Both options are fine with me. I will admit that I'm not as fluent/quick reading code with bitwise operators as I should be, but then again if even the original author of this code got it wrong at first (https://github.com/bitcoin/bitcoin/pull/11456#discussion_r144391264) it's probably not just me.